contracts/test/integration/pcv/CurveSwapper.t.sol

```diff
@@ -0,0 +1,107 @@
+pragma solidity ^0.8.4;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import {Vm} from "../../utils/Vm.sol";
+import {DSTest} from "../../utils/DSTest.sol";
+import {CurveSwapper} from "../../../pcv/curve/CurveSwapper.sol";
+import {Core} from "../../../core/Core.sol";
+import {MainnetAddresses} from "../fixtures/MainnetAddresses.sol";
+
+contract CurveSwapperIntegrationTest is DSTest {
+    Vm public constant vm = Vm(HEVM_ADDRESS);
+    address payable receiver = payable(address(3));
+
+    Core private core = Core(MainnetAddresses.CORE);
+
+    CurveSwapper private curveSwapper;
+
+    // dummy function to avoid errors
+    // the CurveSwapper will call deposit() on the
+    // tokenReceivingAddress after performing a swap
+    function deposit() public {}
+
+    // Deploy
+    function setUp() public {
+        curveSwapper = new CurveSwapper(
+            address(core), // core
+            MainnetAddresses.CURVE_LUSD_METAPOOL, // curvePool
+            MainnetAddresses.LUSD, // tokenSpent
+            MainnetAddresses.DAI, // tokenReceived
+            address(this), // tokenReceivingAddress
+            10000 // maxSlippageBps
+        );
+
+        vm.label(address(this), "test");
+        vm.label(MainnetAddresses.CURVE_LUSD_METAPOOL, "CURVE_LUSD_METAPOOL");
+        vm.label(MainnetAddresses.LUSD, "LUSD");
+        vm.label(MainnetAddresses.DAI, "DAI");
+        vm.label(MainnetAddresses.GUARDIAN_MULTISIG, "GUARDIAN_MULTISIG");
+        vm.label(MainnetAddresses.FEI_DAO_TIMELOCK, "FEI_DAO_TIMELOCK");
+    }
+
+    // Check deployed contract & getters for public properties
+    function testInit() public {
+        assertEq(curveSwapper.curvePool(), MainnetAddresses.CURVE_LUSD_METAPOOL);
+        assertEq(curveSwapper.tokenSpent(), MainnetAddresses.LUSD);
+        assertEq(curveSwapper.tokenReceived(), MainnetAddresses.DAI);
+        assertEq(curveSwapper.maxSlippageBps(), 10000);
+        assertEq(curveSwapper.tokenReceivingAddress(), address(this));
+        assertEq(curveSwapper.i(), 0);
```

Comment:
Are we confident in the index calculation logic and we don't need to test the calculation for different assets and pools? Really like the tests overall

Reply:
I'd expect we check that with a small integration test or proposal validation script before we deploy a new CurveSwapper ?

Reply:
passed i & j as constructor params